src/plugins/home/server/services/sample_data/data_sets/util.ts

```diff
@@ -19,56 +67,9 @@ export const getSavedObjectsWithDataSource = (
   dataSourceTitle?: string
 ): SavedObject[] => {
   if (dataSourceId) {
+    const idGenerator = (id: string) => `${dataSourceId}_${id}`;
     return saveObjectList.map((saveObject) => {
-      saveObject.id = `${dataSourceId}_` + saveObject.id;
-      // update reference
-      if (saveObject.type === 'dashboard') {
-        saveObject.references.map((reference) => {
-          if (reference.id) {
-            reference.id = `${dataSourceId}_` + reference.id;
-          }
-        });
-      }
-
-      // update reference
-      if (saveObject.type === 'visualization' || saveObject.type === 'search') {
-        const searchSourceString = saveObject.attributes?.kibanaSavedObjectMeta?.searchSourceJSON;
-        const visStateString = saveObject.attributes?.visState;
-
-        if (searchSourceString) {
-          const searchSource = JSON.parse(searchSourceString);
-          if (searchSource.index) {
-            searchSource.index = `${dataSourceId}_` + searchSource.index;
-            saveObject.attributes.kibanaSavedObjectMeta.searchSourceJSON = JSON.stringify(
-              searchSource
-            );
-          }
-        }
-
-        if (visStateString) {
-          const visState = JSON.parse(visStateString);
-          const controlList = visState.params?.controls;
-          if (controlList) {
-            controlList.map((control) => {
-              if (control.indexPattern) {
-                control.indexPattern = `${dataSourceId}_` + control.indexPattern;
-              }
-            });
-          }
-          saveObject.attributes.visState = JSON.stringify(visState);
-        }
-      }
-
-      // update reference
-      if (saveObject.type === 'index-pattern') {
```

Comment:
why removing the reference for index pattern?

Reply:
@wanglam Is it expected? If not, can we fix it? https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6850

Reply:
I think that's removed by accident. Will add them back and add some unit tests.